bench-exchange/src/bench.rs

```diff
@@ -108,6 +150,9 @@ where
         .drain(0..accounts_in_groups)
         .map(Arc::new)
         .collect();
+
+    info!("Generating {:?} keys", total_keys);
```

Comment:
Are we generating the keypairs twice?

Reply:
If we are using drone the keypair will be generated twice (but total number of keypairs will be same as before). If we are using Genesis block to fund the signers accounts, the keypair will be generated once (but only for src and profit accounts).

Reply:
Not quite following, `total_keys` is the same through this code flow.  What is the purpose of this second call to `generate_keypairs` on line 155?

Reply:
The intended flow is this:

Write to client_id_and_stake file (which gets fed into the genesis block):
1.  `write_to_client_file` is true
2. `total_keys` number of keys are generated at line 111. The keys and `fund_amount` are serialized into a yaml file. The function returns from there.
3. bench-exchange client exits at the end of the call. This call flow is used only to generate the yaml file.

Read from client_id_and_stake_file and start full bench-exchange client:
1. `read_from_client_file` is true
2. Code block at line 128 reads the yaml file and gets the funded keypairs. (trader_signers, and swapper_signers are funded in the genesis block).
3. Line 155 creates `total_keys` keypairs that used for `src_pubkeys` and `profile_pubkeys`

`total_keys` value in the original code was  `accounts_in_groups * 4`. Now it is `accounts_in_groups * 2`.

Thoughts?

Reply:
I see, that `return` threw me.  Be nice to see that code path separated out into a different top level function and not as an exit hatch in main code flow.

Also, might be more readable if instead of `keypairs` they were named something like `signer_keypairs` and `account_keypairs`.  That's a nit though so take if or leave it :-)

Reply:
I'll take it :)